components/proxy/src/main/java/com/hotels/styx/admin/handlers/CurrentRequestsHandler.java

```diff
@@ -0,0 +1,163 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.admin.handlers;
+
+import static com.google.common.net.MediaType.PLAIN_TEXT_UTF_8;
+import static com.hotels.styx.api.HttpHeaderNames.CONTENT_TYPE;
+import static com.hotels.styx.api.HttpResponseStatus.OK;
+import static java.lang.management.ManagementFactory.getThreadMXBean;
+
+import java.lang.management.LockInfo;
+import java.lang.management.MonitorInfo;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.nio.charset.Charset;
+
+import com.hotels.styx.api.FullHttpResponse;
+import com.hotels.styx.api.HttpRequest;
+import com.hotels.styx.api.HttpResponse;
+import com.hotels.styx.common.http.handler.BaseHttpHandler;
+import com.hotels.styx.server.track.CurrentRequestTracker;
+
+/**
+ * Admin handler that will help in tracking only the current HTTP requests to Styx.
+ */
+public class CurrentRequestsHandler extends BaseHttpHandler {
+
+    private static final Charset UTF_8 = Charset.forName("UTF-8");
+    private final ThreadMXBean threadMXBean;
+
+    public CurrentRequestsHandler() {
+        this.threadMXBean = getThreadMXBean();
+    }
+
+    @Override
+    public HttpResponse doHandle(HttpRequest request) {
+        boolean withStackTrace = request.queryParam("withStackTrace").isPresent() && "true".equalsIgnoreCase(request.queryParam("withStackTrace").get());
+        return FullHttpResponse
+                .response(OK)
+                .disableCaching()
+                .header(CONTENT_TYPE, PLAIN_TEXT_UTF_8)
+                .body(getCurrentRequestContent(withStackTrace), UTF_8, true)
+                .build()
+                .toStreamingResponse();
+    }
+
+    private String getCurrentRequestContent(boolean withStackTrace) {
+        StringBuilder sb = new StringBuilder();
+        CurrentRequestTracker.INSTANCE.getCurrentRequests().forEach(req -> {
+            sb.append("[\n");
+            sb.append(req.getRequest().replaceAll(",", "\n"));
+            sb.append("\n\n");
+            sb.append("running for: ");
+            sb.append(System.currentTimeMillis() - req.getStartingTimeMillies());
+            sb.append("ms");
+            sb.append("\n\n   ...   Thread Info: ...\n");
+            if ("WAITING_FOR_RESPONSE".equalsIgnoreCase(req.getState())) {
```

Comment:
See my other comment in `StandardHttpPipeline.proceed` method about pipeline handler state not being a sufficient condition for request being sent. It might still be held by some asynchronous plugin in a separate thread.

So you would have to change this part of the code.

However I would like to see the HTTP pipeline handler state as an additional piece of information:

```
    Styx server state: WAITING_FOR_RESPONSE
    running for: 1ms

    ... Thread Info: ...
    [stack trace for some asynchronous plugin]
```